python/arcticdb/version_store/_normalization.py

```diff
@@ -497,6 +514,20 @@ def _index_to_records(self, df, pd_norm, dynamic_strings, string_max_len):
             df.reset_index(fields, inplace=True)
             index = df.index
         else:
+            n_rows = len(index)
+            n_categorical_columns = len(df.select_dtypes(include="category").columns)
+            if IS_PANDAS_TWO and isinstance(index, (Index, RangeIndex)) and n_rows == 0 and n_categorical_columns == 0:
```

Comment:
Both `RangeIndex` and `DatetimeIndex` are sub-types of `Index`, so the `isinstance(index, (Index, RangeIndex))` part is probably always True.